density/static/map.js

```diff
@@ -0,0 +1,103 @@
+// vanilla js document.ready bc we are kool kids
+document.addEventListener('DOMContentLoaded', function() {
+  document.getElementById('map').addEventListener('load', function() {
+    
+    //Iterate through locations and display their data
+    var data = document.querySelector('#data');
+    var locations = JSON.parse(data.dataset.locations);
+    var map = document.getElementById('map');
+    var innerSvg = map.contentDocument;
+    var popup = document.getElementById('locationPopup');
+
+    var initialOffsetTop = map.getBoundingClientRect().top + window.scrollY;
+    var initialOffsetLeft = map.getBoundingClientRect().left + window.scrollX;
+    
+    //Create a mapping from parent_id --> text representation
+    var parentToText = {};
+    
+    //Create a mapping from parent_id --> list full %s for elements inside (e.g. floors of a building)
+    var parentToList = {};
+
+    //Create a mapping from parent_id --> query (http://....hours)
+    var parentToQuery = {};
+    
+    for (var i = 0; i < locations.length; i++) { 
+      if (locations[i].name != 'Butler Library stk') { //We're skipping this out of non-use]
+        if (parentToList[locations[i].parentId] == null) {
```

Comment:
this whole loop looks like it could be more readable by pulling `locations[i]` out into a variable